### DISCUSSION TITLE
Upgrading to v4, attempting to remove zustand/context

### DISCUSSION THREAD

**Discussion by evancloutier** [pmndrs/zustand#1171] [Q&A]
Hello! I'm attempting to upgrade to v4, and in the process of that I'd like to get rid of my usage of `zustand/context`.

Am running into an interesting issue when initializing my store with slices. I'm on the latest version of NextJS (v12.2.4), but I also experienced this on v12.1.0.

If I initialize the state from my slice within the store creation method, everything will work as intended:

```typescript
interface ModalSlice {
  open: boolean;
}

type StoreState = ModalSlice; // Eventually there will be more slices!

const useStore = create<StoreState>()((...a) => ({
  open: false
}))
```

However, if I start to break the state into slices:

```typescript
// store/index.ts
import { ModalSlice, createModalSlice } from "./store/modal-slice";

type StoreState = ModalSlice; // Eventually there will be more slices!

const useStore = create<StoreState>()((...a) => ({
  ...createModalSlice(...a)
}))

// store/modal-slice/index.ts

export const createModalSlice: StateCreator<
  StoreState,
  [],
  [],
  ModalSlice
> = _ => ({
  open: false,
});
```

I'll run into an odd error about webpack imports:

```TypeError: _modal_slice__WEBPACK_IMPORTED_MODULE_0__.createModalSlice is undefined```

<img width="954" alt="Screen Shot 2022-08-05 at 11 07 18 AM" src="https://user-images.githubusercontent.com/9866928/183106533-e6077389-94d7-49cc-ab45-87251b96af86.png">

Any thoughts on how to alleviate this issue? I would like the ability to split into slices. I have attempted enabling/disabling `esmExternals` in my `next.config.js` but it seems it does not make a difference.

**Comment by dai-shi** [comment]
The error seems something related with bundling, and not sure if it's related with zustand/context or even zustand v4.

Does it work if you put `createModalSlice` in the same file (`store/index.ts`)?

**Reply by evancloutier** [reply]
> The error seems something related with bundling, and not sure if it's related with zustand/context or even zustand v4.
> 
> Does it work if you put `createModalSlice` in the same file (`store/index.ts`)?

This does fix the issue – seems unrelated to zustand and is a bundling issue of some sort. I will see if I can get a consistent recreation and report back.
